### PR TITLE
[Klima data] Fix for mobile menu

### LINKED ISSUE #1761: [klima data - mobile] hamburger menu not functional
The hamburger menu button in the top right corner on all sites is not working at all, tapping it does not do anything. Tested on Chromium-based browsers and Firefox.

### LINKED ISSUE #1796: [klima data] language selector popup does not close automatically
When I click the language button and **don't** select a language, the dropdown menu does not close automatically if I click anywhere else on the site. To close it, I have to select a language or press the button again. Pretty inconvenient.

On KlimaDAO and Carbonmark, the menu closes automatically if I click something else on the site. For consistency's sake, this should behave the same on KlimaData, imo. 

Same on mobile, btw.

### PR DESCRIPTION
**Pull request by biwano**
## Description

- Adds a mobile menu
- Refactors layout components
- Hides the language button

## Related Ticket

Resolves #1761 
Resolves #1796

## Changes

| Before  | After  |
|---------|--------|
|img here |img here|

## Checklist

- [X] I have run `npm run build-all` without errors
- [X] I have formatted JS and TS files with `npm run format-all`

### CODE REVIEW

**Comment by 0xtapi** [comment]
Hey there, a few issues:

Sometimes, but not always, clicking the first link in the menu leads to this:

Screenshot_20231101-123153-390

Also, sometimes the link stops working. It only happens on the first link ("Overview"), and not every time, and I can't figure out why sometimes nothing happens and sometimes it throws the error message. Not sure how to reproduce this one consistently. Accessing the console on mobile browser apparently isn't so straight forward (need to hook the phone up to a PC/Mac). If you need the console printout, I'll look into this later/tomorrow (bit of a packed day for me).

Also, the "constantly redraw charts" bug is in this one. Not sure if by accident or by design because your fix hasn't been merged yet. If it's the latter, you can ignore this one.

**Comment by sprrwhwk** [comment]
Hey, the menu looks great but it doesn't match the designs: https://www.figma.com/file/mXI4SzNNmoefYnOdcbXZcO/Carbon-Dashboard?node-id=3086%3A10114&mode=dev

Are there other designs for the menu?

**Comment by biwano** [comment]
> Hey there, a few issues:
> 
> Sometimes, but not always, clicking the first link in the menu leads to this:
> 
> Screenshot_20231101-123153-390
> 
> Also, sometimes the link stops working. It only happens on the first link ("Overview"), and not every time, and I can't figure out why sometimes nothing happens and sometimes it throws the error message. Not sure how to reproduce this one consistently. Accessing the console on mobile browser apparently isn't so straight forward (need to hook the phone up to a PC/Mac). If you need the console printout, I'll look into this later/tomorrow (bit of a packed day for me).
> 
> Also, the "constantly redraw charts" bug is in this one. Not sure if by accident or by design because your fix hasn't been merged yet. If it's the latter, you can ignore this one.

I could reproduce it. Another tricky one I fear. On it.

**Comment by biwano** [comment]
@sprrwhwk @0xtapi The bug with the overview link not working properly was beyond tricky, I had to make heavier changes than I wanted. Let me know if you see anything weird.

File: carbon/next.config.js [hunk 1]
```diff
@@ -12,22 +12,6 @@ module.exports = async (phase, { defaultConfig }) => {
...
-        {
-          source: "/:locale/token-details",
-          destination: "/:locale/token-details",
-        },
-        {
-          source: "/:locale/retirement_trends",
-          destination: "/:locale/retirement_trends/by-pool",
-        },
-      ];
-    },
```
Comment (sprrwhwk) [inline carbon/next.config.js:30]: What does this change achieve?

Comment (biwano) [reply carbon/next.config.js:30]: It removes the rewrites. There is an optimization in nextJS that replaces internal links so the page is not reloaded and the site works as a single page app. But I found out that this breaks with rewrites. That's why the Overview link was not working sometimes.
So the first one is replaced by the file "app/[locale]/page.tsx"
And the other two were no longer used.

**Review by sprrwhwk — APPROVED** [review]

**Comment by 0xtapi** [comment]
Don't hate the messenger please, but I found another small issue:

3e961559-88c3-45fd-af48-459070e5bdd8


This is what this particular site (Retirement Trends) looks like on Firefox. Menu opens at the bottom and there's a small transparent element on top of the menu. All other sites look good (and work) on all other browsers. Not really a show stopper, but maybe you can take a look again and see if there's a quick fix for this?

**Review by jabby09 — APPROVED** [review]

**Pull request merged**